tests/conftest.py

```diff
@@ -220,6 +220,44 @@ def _create_profile(name, **kwargs):
     return _create_profile
 
 
+@pytest.fixture
+def create_config_instance(create_empty_config_instance, create_profile) -> Config:
+    """Create a temporary configuration instance with one profile.
+    This creates a temporary directory with a clean `.aiida` folder and basic configuration file.
+    The currently loaded configuration and profile are stored in memory,
+    and are automatically restored at the end of this context manager.
+    This fixture differs from `create_empty_config_instance` in that here a profile will be created and set as default.
+    The defaults of the profile can be overridden in the callable, as well as whether it should be set as default.
+    This fixture should probably be used for most tests that assume a basically configured setup.
```

Comment:
Can you expand on this a bit?

In particular, I think one should contrast this to the `aiida_profile` fixture that is used automatically by all tests (`autouse=True`) and will already have set up a profile and configuration

Reply:
Note I copied this from @sphuber's #4245, so maybe I could defer to him on that 😉

Reply:
The idea of the fixture is that it should be used for tests that manipulate the config. For those tests we don't want to manipulate the _actual_ config that is being used for the test suite, but we want a dummy one. Otherwise, if the test fails and doesn't undo its changes properly, the actual config is broken and all other tests fail.

Reply:
cheers for the clarification